server/stream.go

```diff
@@ -2385,7 +2385,6 @@ func (mset *stream) internalLoop() {
 
 				// Do this here to nil out below vs up in for loop.
 				next := pm.next
-				pm.next, pm.hdr, pm.msg = nil, nil, nil
 				pm = next
```

Comment:
Not entirely sure what's going on here. Seems like niling out of `pm` before `pm = next` is unnecessary because the last line overwrites the address.